### ISSUE TITLE
Stat : provide monthly active users statistic

### ISSUE THREAD

**Issue by yuriyz** [GluuFederation/oxAuth#1512]
## Describe the issue

Goal is to provide monthly active users. Idea is to implement it with HLL help as following:

1. Each server should persists HLL data to persistence each 15min (or some other interval, configurable?).
2. Entry in persistence should be : per server and per month (means entry data has to be updated for same server and same month).
3. Monthly report means : HLL union for all servers/nodes for given month. If server wrote HLL data and after 5min crashed then we lost data for these 5min.

Java implementation: `net.agkn.hll`.

10 servers with `1000000` unique users login (total 10M) log2m=15, regwidth = 5
```
server0 - size: 20483bytes, cardinality: 996522
server1 - size: 20483bytes, cardinality: 998560
server2 - size: 20483bytes, cardinality: 1001721
server3 - size: 20483bytes, cardinality: 988345
server4 - size: 20483bytes, cardinality: 998151
server5 - size: 20483bytes, cardinality: 1005606
server6 - size: 20483bytes, cardinality: 993774
server7 - size: 20483bytes, cardinality: 992465
server8 - size: 20483bytes, cardinality: 988403
server9 - size: 20483bytes, cardinality: 1002353
UNION - size: 20483bytes, cardinality: 10010897
```

**Comment by yuriyz** [comment]
Stat data aggregation is performed by web service which is protected and has time request limit of 60 seconds by default (configurable by `statWebServiceIntervalLimitInSeconds`).

Sample request (single month)
```
GET /oxauth/restv1/stat?month=202101
Authorization: Basic czZCaGRSa3F0MzpnWDFmQmF0M2JW
```

Sample request (multiple months, e.g. 202012 (2020 Dec) 202101 (2021 Jan))
```
GET /oxauth/restv1/stat?month=202012%20202101
Authorization: Basic czZCaGRSa3F0MzpnWDFmQmF0M2JW
```

Sample Response
```
{
  "response": {
    "202101": {
      "monthly_active_users": 1939497078,
      "token_count_per_granttype": {
        "implicit": {
          "access_token": 8843443,
          "refresh_token": 361678620,
          "uma_token": 257653798,
          "id_token": 1196937417
        },
        "refresh_token": {
          "access_token": 198484847,
          "refresh_token": 527458534,
          "uma_token": 1552211159,
          "id_token": 2077968838
        },
        "password": {
          "access_token": 21293868,
          "refresh_token": 2003904704,
          "uma_token": 44144850,
          "id_token": 1713870170
        },
        "client_credentials": {
          "access_token": 1656091541,
          "refresh_token": 875245682,
          "uma_token": 1920358732,
          "id_token": 2029924857
        },
        "urn:ietf:params:oauth:grant-type:device_code": {
          "access_token": 871050693,
          "refresh_token": 735390384,
          "uma_token": 902227655,
          "id_token": 572398177
        },
        "authorization_code": {
          "access_token": 1468057290,
          "refresh_token": 1576996227,
          "uma_token": 1755565333,
          "id_token": 944346498
        },
        "urn:openid:params:grant-type:ciba": {
          "access_token": 1187334485,
          "refresh_token": 1043290537,
          "uma_token": 206572517,
          "id_token": 1850166398
        },
        "urn:ietf:params:oauth:grant-type:uma-ticket": {
          "access_token": 933970698,
          "refresh_token": 1089756841,
          "uma_token": 352343374,
          "id_token": 645686974
        }
      }
    },
    "202012": {
      "monthly_active_users": 1620607010,
      "token_count_per_granttype": {
        "implicit": {
          "access_token": 20676511,
          "refresh_token": 960308503,
          "uma_token": 703633153,
          "id_token": 573544490
        },
        "refresh_token": {
          "access_token": 526463917,
          "refresh_token": 483679318,
          "uma_token": 545888615,
          "id_token": 951286042
        },
        "password": {
          "access_token": 1883167951,
          "refresh_token": 1636169697,
          "uma_token": 1753544837,
          "id_token": 1675535757
        },
        "client_credentials": {
          "access_token": 290209545,
          "refresh_token": 1635984977,
          "uma_token": 1066472511,
          "id_token": 463520941
        },
        "urn:ietf:params:oauth:grant-type:device_code": {
          "access_token": 84124820,
          "refresh_token": 1705729386,
          "uma_token": 2070635541,
          "id_token": 970034925
        },
        "authorization_code": {
          "access_token": 387026047,
          "refresh_token": 907425179,
          "uma_token": 370523863,
          "id_token": 1807260512
        },
        "urn:openid:params:grant-type:ciba": {
          "access_token": 1408098991,
          "refresh_token": 819153423,
          "uma_token": 936295536,
          "id_token": 378210748
        },
        "urn:ietf:params:oauth:grant-type:uma-ticket": {
          "access_token": 381038617,
          "refresh_token": 828068056,
          "uma_token": 448746427,
          "id_token": 650520169
        }
      }
    }
  }
}
```

**Comment by yuriyz** [comment]
Implemented in 4.2.3 and jans. QAed for single instance and results are good. It still needs to be QAed for k8s case.

**Comment by yuriyz** [comment]
This feature works in 4.3 and in janssen. It does not work correctly in 4.2.3.
